### ISSUE TITLE
Cannot run the compiled executable on linux

### ISSUE THREAD

**Issue by nboldi** [ELTE-Soft/txtUML#406]
It seems that threading is somehow not enabled:
```
terminate called after throwing an instance of 'std::system_error'
  what():  Enable multithreading to use std::thread: Operation not permitted
Aborted (core dumped)
```

**Comment by nboldi** [comment]
However it might be a buggy ghc version... I'll check it out.

**Comment by nboldi** [comment]
Yes it seems to be g++ bug, discussed here: http://stackoverflow.com/questions/19463602/compiling-multithread-code-with-g

I did a walkaround by modifying the CMakeLists.txt, adding: 
```
  add_compile_options(-Wl,--no-as-needed -lpthread)
```
and
```
  set_target_properties(main PROPERTIES LINK_FLAGS_DEBUG "-Wl,--no-as-needed -lpthread")
  set_target_properties(main PROPERTIES LINK_FLAGS_RELEASE "-Wl,--no-as-needed -lpthread")
```

**Comment by nboldi** [comment]
But as a fix it should depend on the compiler version to insert these walkaround flags.

**Comment by gerazo** [comment]
This https://github.com/ELTE-Soft/txtUML/blob/master/dev/plugins/hu.elte.txtuml.export.cpp/src/hu/elte/txtuml/export/cpp/CMakeSupport.java#L219-L223 should add pthread support if it is found. The reason it is implemented as a conditional operations is that on Windows, it cannot be found but is used by default. On POSIX systems, the library should be found and the flags should be added. Without the extra flags, were the pthread linked to our stuff? If it was on the link line and we still got this problem it means the nobody referred it. Probably someone added an extra library linked later as pthread lib. Linking pthread should be the last library on the link line as the order should be our stuff --- their stuff.

**Comment by gerazo** [comment]
I've updated the server and ensured that the latest compiler and everything is installed.

**Comment by nboldi** [comment]
I've just updated my ubuntu to have the latest gcc available. The bug is still present in gcc 4.8.4

**Comment by nboldi** [comment]
The `-lpthread` flag might be unnecessary, since it is already in the list of libraries, however the `-Wl,--no-as-needed` flags are part of the walkaround.

**Comment by gerazo** [comment]
Which branch is this? I can't see a problem on Jenkins. Jenkins has gcc 5.4

**Comment by gerazo** [comment]
Can you send me the link command which runs for this executable?

**Comment by gerazo** [comment]
One which seems to currently work is: [19/19] : && /usr/bin/g++   -O3 -DNDEBUG  -flto CMakeFiles/main.dir/main.cpp.o CMakeFiles/main.dir/deployment.cpp.o CMakeFiles/main.dir/associations.cpp.o CMakeFiles/main.dir/Engine.cpp.o CMakeFiles/main.dir/Backwards_subSM.cpp.o CMakeFiles/main.dir/Forwards_subSM.cpp.o CMakeFiles/main.dir/Gearbox.cpp.o CMakeFiles/main.dir/Lamp.cpp.o  -o main  -rdynamic liblibsmrt.a -lpthread
